src/runtime_src/core/include/xgq_impl.h

```diff
@@ -232,15 +231,18 @@ static inline bool xgq_can_consume(struct xgq *xgq)
  * 	  -> xgq_consume() -> process-CQ-entry -> xgq_notify_peer_consumed()
  *
  * Server -> xgq_attach() -> xgq_consume() -> process-SQ-entry -> xgq_notify_peer_consumed()
- * 	  -> xgq_produce() -> fill-up-CQ-entry -> xgq_notify_peer_consumed()
+ * 	  -> xgq_produce() -> fill-up-CQ-entry -> xgq_notify_peer_produced()
+ *
+ * You can call xgq_produce() and fill out entries multiple times before you call
+ * xgq_notify_peer_produced(), which then will publish all entries at once to peer.
  */
 
 static inline int xgq_alloc(struct xgq *xgq, bool server, uint64_t io_hdl, uint64_t ring_addr,
 	    size_t *ring_len, uint32_t slot_size, uint64_t sq_produced, uint64_t cq_produced)
 {
 	size_t rlen = *ring_len;
 	uint32_t numslots = XGQ_MIN_NUM_SLOTS;
-	struct xgq_header hdr = { 0 };
+	struct xgq_header hdr;
```

Comment:
warning: uninitialized record type: `hdr` [cppcoreguidelines-pro-type-member-init,hicpp-member-init]

```cpp
struct xgq_header hdr;
^
                     {}
```

Reply:
clang-tidy complains about {0}, so this has to be removed. can we tell clang-tidy to ignore this line?

Reply:
It is clangtidy that added the comment above and the variable should be initialized for sure. You can do just `struct xgq_header hdr {};`  but interestingly, when I run clangtidy  it doesn't complain about = {0}.

Reply:
added struct xgq_header hdr = {}; see if this is OK now?